### PR TITLE
Feat/update actor

### PR DESCRIPTION
**Pull request by connorsmith256**
## Feature or Problem
This PR adds the ability to update/replace running actors with an atomic command



## Related Issues
Resolves https://github.com/wasmCloud/wasmCloud/issues/374

## Release Information
Next

## Consumer Impact
Users can now use `wash ctl update actor` with the Rust host

## Testing
<!---
Declare the testing information for this pull request
--->

<!---
Identify the platforms on which this code was built (include both OS and CPU architecture)
--->
Built on platform(s)
- [ ] x86_64-linux
- [ ] aarch64-linux
- [ ] x86_64-darwin
- [ ] aarch64-darwin
- [ ] x86_64-windows

<!---
Identify the platforms on which this code was tested (include both OS and CPU architecture)
--->
Tested on platform(s)
- [ ] x86_64-linux
- [ ] aarch64-linux
- [ ] x86_64-darwin
- [ ] aarch64-darwin
- [ ] x86_64-windows

### Unit Test(s)
<!---
Indicate if unit tests were added or modified, and if so, which ones 
--->

### Acceptance or Integration
I didn't write a new integration test. In order to test this manually (see below), I relied on having two actors built  with the same public key but different implementation details. So in order to include an automated test of this fashion, I would need to add two test fixture actors and reproduce the manual test below. I think we'll want that eventually, but for now I'm okay leaving it out. 

### Manual Verification

```
wash ctl start actor -x default --host-id NDYEPXYQTKD3GAFCCCU2HKCC3FEXW23WH4LPX2LSOSHDNS4REUXDJYAI oci://localhost:5000/v2/hello1:0.1.0

>>> ⢀⠀  Starting actor oci://localhost:5000/v2/hello1:0.1.0 ...                                                                                                                                                                     
Actor [MBLRZ3JJ7GVEYHZ6T5VLHZG4WJAEO26YOQBBKG6HEPHNUBZ2P7MAKYKR] (ref: [oci://localhost:5000/v2/hello1:0.1.0]) started on host [NDYEPXYQTKD3GAFCCCU2HKCC3FEXW23WH4LPX2LSOSHDNS4REUXDJYAI]

curl localhost:8080/echo
Hello, World 1!%                                                                                                                                                                                                                    

wash ctl update actor NDYEPXYQTKD3GAFCCCU2HKCC3FEXW23WH4LPX2LSOSHDNS4REUXDJYAI MBLRZ3JJ7GVEYHZ6T5VLHZG4WJAEO26YOQBBKG6HEPHNUBZ2P7MAKYKR oci://localhost:5000/v2/hello2:0.1.0

Actor MBLRZ3JJ7GVEYHZ6T5VLHZG4WJAEO26YOQBBKG6HEPHNUBZ2P7MAKYKR updated to oci://localhost:5000/v2/hello2:0.1.0

curl localhost:8080/echo
Hello, World 2!%       
```

### CODE REVIEW

**Review by thomastaylor312 — APPROVED** [review]

File: crates/host/src/wasmbus/mod.rs [hunk 1]
```diff
@@ -794,13 +807,12 @@ impl Lattice {
         Ok(entry.insert(actor))
     }
 
-    #[instrument(skip(host_id))]
+    #[instrument(skip(self))]
     async fn stop_actor<'a>(
         &self,
         entry: hash_map::OccupiedEntry<'a, String, Arc<Actor>>,
-        host_id: impl AsRef<str>,
+        host_id: &str,
```
Comment (rvolosatovs) [inline crates/host/src/wasmbus/mod.rs:814]: why?

Comment (thomastaylor312) [reply crates/host/src/wasmbus/mod.rs:814]: As far as I can tell looking at the code, everything calling this private function has the ability to just use a normal `str` (or deref-ed `String`). Plus it has the added advantage of simplifying the logging

**Comment by connorsmith256** [comment]
@rvolosatovs next time can you let me decide when to merge my PR? It's technically part of [our PR lifecycle rules](https://github.com/wasmCloud/wasmCloud/blob/main/CONTRIBUTING.md#pr-lifecycle)

**Comment by rvolosatovs** [comment]
> @rvolosatovs next time can you let me decide when to merge my PR? It's technically part of [our PR lifecycle rules](https://github.com/wasmCloud/wasmCloud/blob/main/CONTRIBUTING.md?rgh-link-date=2023-08-04T20%3A50%3A13Z#pr-lifecycle)

Sure - I did not expect an approved "ready-to-review" PR not to be ready for merge

**Pull request merged**